### PR TITLE
Better Calendar Widget Data

### PR DESCRIPTION
**Pull request by miaulalala**
Needs https://github.com/nextcloud/server/pull/36620

Resolved part of https://github.com/nextcloud/calendar/issues/4813

Signed-off-by: Anna Larch <anna@nextcloud.com>

### CODE REVIEW

File: lib/Dashboard/CalendarWidget.php [hunk 1]
```diff
@@ -143,31 +154,65 @@ public function load(): void {
...
+				if(!empty($dtEnd) && $dtStart->diff($dtEnd)->days >= 1) {
+					// All day (and longer)
+					$timeString = $this->dateTimeFormatter->formatDateSpan($dtStart);
+				} elseif(!empty($dtEnd)) {
+					$timeString = $this->dateTimeFormatter->formatDateTime($dtStart, 'short', 'short') . ' - ' . $this->dateTimeFormatter->formatTime($dtEnd, 'short');
+				} else {
+					$timeString = $this->dateTimeFormatter->formatTimeSpan($dtStart, $startTimeWithTimezone);
+				}
 				$widget = new WidgetItem(
 					$calendarEvent['objects'][0]['SUMMARY'][0] ?? 'New Event',
```
Comment (tcitworld) [inline lib/Dashboard/CalendarWidget.php:211]: **thought (non-blocking):** Unrelated, but the fallback should rather be named `Untitled event`.

File: lib/Dashboard/CalendarWidget.php [hunk 2]
```diff
@@ -143,31 +154,65 @@ public function load(): void {
...
+		//          const start = dateFactory()
+		//			const end = dateFactory()
+		//			end.setDate(end.getDate() + 14)
+		//          const startOfToday = moment(start).startOf('day').toDate()
+		// get all vevents in this time range
+		// if "show tasks" is enabled, get all todos in the time range
+		// sort events by time
+		// filter events by:
+		// .filter(event => !event.classNames.includes('fc-event-nc-task-completed'))
+		// .filter(event => !event.classNames.includes('fc-event-nc-cancelled'))
```
Comment (tcitworld) [inline lib/Dashboard/CalendarWidget.php:167]: **issue:** This part is no longer taken into account.

**Review by tcitworld — COMMENTED** [review]
Since the time formatting is now done on the backend's side, we should format datetime using the user's own timezone setting, which might not be the same as the calendar's timezone or event the calendar object start or end time timezone.

File: lib/Dashboard/CalendarWidget.php (hunk 1 above)
Comment (miaulalala) [reply lib/Dashboard/CalendarWidget.php]: right, then it's in line with the frontend. nice catch

File: lib/Dashboard/CalendarWidget.php (hunk 2 above)
Comment (miaulalala) [reply lib/Dashboard/CalendarWidget.php:167]: yoda-patience

File: lib/Dashboard/CalendarWidget.php [hunk 3]
```diff
@@ -30,6 +30,7 @@
 use DateTimeImmutable;
 use OCA\Calendar\AppInfo\Application;
 use OCA\Calendar\Service\JSDataService;
+use OCA\DAV\CalDAV\CalendarImpl;
```
Comment (ChristophWurst) [inline lib/Dashboard/CalendarWidget.php:33]: this is a private API

Comment (ChristophWurst) [reply lib/Dashboard/CalendarWidget.php:33]: yeah, still means this will introduce an inter-app dependency. we should avoid that at all cost.

Comment (tcitworld) [reply lib/Dashboard/CalendarWidget.php:33]: `IGetTimezone` is just fine, right?

**Comment by miaulalala** [comment]
Superceded by https://github.com/nextcloud/calendar/pull/5417

**Pull request closed without merging**